### ISSUE TITLE
Introduce mechanism for translation fallbacks from Translatewiki

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#4496]
When a string's content is changed and its name is changed, we completely lose the context to the string (even though it may have a valid fallback until it's fully translated). Translatewiki doesn't support this out-of-the-box, and we could consider reaching out to them to ask if it's something that they'd be interested in supporting. Otherwise, we might be able to support it by introducing a mechanism that leverages Oppia's own string selection to choose a fallback string if the new string isn't yet available in a particular language.

**Comment by seanlip** [comment]
In general, shouldn't we be keeping the translation key/name the same (even though we change the content of the English string)?

**Comment by BenHenning** [comment]
Not necessarily. In some cases we do UI redesigns which may result in different designations for the same UI element, along with an updated string. I think the process to do this today would be to rename the string (without changing its content), checking that in, waiting for Translatewiki to pick it up & push updates, merge that, then update the translation contents so that the translations can be marked as stale. Changing both at the same time will look like a deletion & addition to Translatewiki.

**Comment by seanlip** [comment]
Ah ok. But then the context should change too, right (if the element now fulfills a different purpose in the updated UI redesign)?

**Comment by BenHenning** [comment]
The context does change, but it's sometimes the case that the old string is reasonable as a stop-gap rather than reverting back to English while we wait for translations.

@JishnuGoyal's recent PR is a good example of this. We're renaming the 'All Topics' header to be 'Select a Topic' (or similar--might not be exact wording). In this case, it's quite nice to defer back to "All Topics", but the context here is technically different (rather than just being a header, it's now instructional). I will admit this isn't a perfect example since there was an initial string name that might have been sufficiently clear "e.g. "home_topic_list_header" but we didn't name it that generically originally, so it's exactly running into this situation.

**Comment by seanlip** [comment]
Ah, I see, interesting, so the idea is to link to another string as the fallback.

Hm I don't know, it feels a bit like niche functionality and I imagine this case shouldn't really happen that often (and, as you mention, the clarity of the string name that plays into this as well). But I guess we could check. Alternatively, I imagine we could do this on our end as well during build time; that might be easier since I think it's somewhat custom to us (it's more a question of how we decide to use the translated strings, so the mapping should probably happen on our end).

**Comment by BenHenning** [comment]
A build time translation of keys is an interesting idea, and might more or less achieve the same thing. More technical design is needed here.
